tools/modules/generate_mettle_payloads.rb

```diff
@@ -12,35 +12,37 @@
   'https'
 ]
 
-arches = [
-  ['aarch64',   'Linux', 'aarch64-linux-musl'],
-  ['armbe',     'Linux', 'armv5b-linux-musleabi'],
-  ['armle',     'Linux', 'armv5l-linux-musleabi'],
-  ['mips64',    'Linux', 'mips64-linux-muslsf'],
-  ['mipsbe',    'Linux', 'mips-linux-muslsf'],
-  ['mipsle',    'Linux', 'mipsel-linux-muslsf'],
-  ['ppc',       'Linux', 'powerpc-linux-muslsf'],
-  ['ppce500v2', 'Linux', 'powerpc-e500v2-linux-musl'],
-  ['ppc64le',   'Linux', 'powerpc64le-linux-musl'],
-  ['x64',       'Linux', 'x86_64-linux-musl'],
-  ['x86',       'Linux', 'i486-linux-musl'],
-  ['zarch',     'Linux', 's390x-linux-musl'],
-  ['x64',       'OSX',   'x86_64-apple-darwin'],
-  ['aarch64',   'Apple_iOS',   'aarch64-iphone-darwin'],
-  ['armle',     'Apple_iOS',   'arm-iphone-darwin'],
+arch_list = [
+  { arch: 'aarch64',   platform: 'Linux', payload: 'aarch64-linux-musl', mixins: ['Msf::Payload::Linux'] },
+  { arch: 'armbe',     platform: 'Linux', payload: 'armv5b-linux-musleabi', mixins: ['Msf::Payload::Linux'] },
+  { arch: 'armle',     platform: 'Linux', payload: 'armv5l-linux-musleabi', mixins: ['Msf::Payload::Linux'] },
+  { arch: 'mips64',    platform: 'Linux', payload: 'mips64-linux-muslsf', mixins: ['Msf::Payload::Linux'] },
+  { arch: 'mipsbe',    platform: 'Linux', payload: 'mips-linux-muslsf', mixins: ['Msf::Payload::Linux'] },
+  { arch: 'mipsle',    platform: 'Linux', payload: 'mipsel-linux-muslsf', mixins: ['Msf::Payload::Linux'] },
+  { arch: 'ppc',       platform: 'Linux', payload: 'powerpc-linux-muslsf', mixins: ['Msf::Payload::Linux'] },
+  { arch: 'ppce500v2', platform: 'Linux', payload: 'powerpc-e500v2-linux-musl', mixins: ['Msf::Payload::Linux'] },
+  { arch: 'ppc64le',   platform: 'Linux', payload: 'powerpc64le-linux-musl', mixins: ['Msf::Payload::Linux'] },
+  { arch: 'x64',       platform: 'Linux', payload: 'x86_64-linux-musl', mixins: ['Msf::Payload::Linux'] },
+  { arch: 'x86',       platform: 'Linux', payload: 'i486-linux-musl', mixins: ['Msf::Payload::Linux'] },
+  { arch: 'zarch',     platform: 'Linux', payload: 's390x-linux-musl', mixins: ['Msf::Payload::Linux'] },
+  { arch: 'x64',       platform: 'OSX',   payload: 'x86_64-apple-darwin', mixins: ['Msf::Payload::Osx'] },
+  { arch: 'aarch64',   platform: 'Apple_iOS',   payload: 'aarch64-iphone-darwin', mixins: [] },
+  { arch: 'armle',     platform: 'Apple_iOS',   payload: 'arm-iphone-darwin', mixins: [] },
 ]
 
 arch = ''
 payload = ''
 platform = ''
+mixins = ''
 scheme = ''
 cwd = File::dirname(__FILE__)
 
-arches.each do |a, pl, pa|
+arch_list.each do |arch_hash|
   schemes.each do |s|
-    arch = a
-    platform = pl
-    payload = pa
+    arch = arch_hash[:arch]
+    platform = arch_hash[:platform]
+    payload = arch_hash[:payload]
+    mixins = arch_hash[:mixins].join("\n  include ")
```

Comment:
Normally best practice involves moving all 'view' logic into the erb template, and keeping it out of the 'controller'. i.e. joining arrays with strings etc isn't normally the go-to approach, instead there'd be a loop in erb 🔁